app/src/main/java/com/scouts/kitchenplaner/ui/view/createproject/AllergenAdder.kt

```diff
@@ -41,50 +45,66 @@ import androidx.compose.runtime.setValue
-    val columnItemModifier = Modifier
-        .padding(5.dp)
-        .height(70.dp)
-    Column (
-        modifier = Modifier.verticalScroll(rememberScrollState())
-    ) {
-        AllergenInputs(state, columnItemModifier)
+fun AllergenAdder(state: AllergenPersonAdderState, onAdd: () -> Unit, onDismiss: () -> Unit) {
+    Dialog(onDismissRequest = onDismiss) {
+        Surface(
+            modifier = Modifier.fillMaxHeight(0.5f),
+            shape = RoundedCornerShape(16.dp),
+            color = MaterialTheme.colorScheme.background
+        ) {
+            val columnItemModifier = Modifier
+                .padding(5.dp)
+                .height(70.dp)
+            Column(
+                modifier = Modifier
+                    .verticalScroll(rememberScrollState())
+            ) {
+                AllergenInputs(state, columnItemModifier)
 
-        HorizontalDivider()
+                HorizontalDivider()
 
-        AllergenAddInputs(
-            onAdd = { allergen, traces -> state.addAllergen(allergen, traces) },
-            modifier = Modifier.padding(10.dp)
-        )
-
-        if (state.allergens.isNotEmpty()) {
-            HorizontalDivider()
+                AllergenAddInputs(
+                    onAdd = { allergen, traces -> state.addAllergen(allergen, traces) },
+                    modifier = Modifier.padding(10.dp)
+                )
 
-            Column {
-                state.allergens.forEach { (allergen, traces) ->
-                    val text = if (traces) {" (Spuren)"} else ""
-                    Text(modifier = Modifier.padding(10.dp), text = "$allergen$text")
+                if (state.allergens.isNotEmpty()) {
+                    HorizontalDivider()
+
+                    Column {
+                        state.allergens.forEach { (allergen, traces) ->
+                            val text = if (traces) {
+                                " (Spuren)"
+                            } else ""
+                            Text(modifier = Modifier.padding(10.dp), text = "$allergen$text")
+                        }
+                    }
                 }
-            }
-        }
-
-        HorizontalDivider()
 
-        IconButton(
-            onClick = {
-                if (state.name != "" && state.allergens.isNotEmpty()) {
-                    onAdd()
+                HorizontalDivider()
+
+                IconButton(
+                    onClick = {
+                        if (state.name != "" && state.allergens.isNotEmpty()) {
+                            onAdd()
+                            onDismiss()
+                        }
+                    },
+                    modifier = Modifier
+                        .padding(start = 10.dp)
+                        .align(Alignment.CenterHorizontally)
+                ) {
+                    Icon(
+                        imageVector = Icons.Filled.AddCircle,
+                        contentDescription = "Add allergen for person"
```

Comment:
-  Description does not describe properly (Add intolerant person). Change Icon to Check mark
- If the person already exists the contend is saved in the dialog? Is this wanted?